processor/src/main/java/org/mapstruct/ap/internal/model/ObjectFactoryMethodResolver.java

```diff
@@ -58,18 +77,18 @@ public static MethodReference getFactoryMethod( Method method,
                 method,
                 getAllAvailableMethods( method, ctx.getSourceModel() ),
                 java.util.Collections.<Type> emptyList(),
-                targetType.getEffectiveType(),
+                alternativeTarget,
                 SelectionCriteria.forFactoryMethods( selectionParameters ) );
 
         if (matchingFactoryMethods.isEmpty()) {
-            return findBuilderFactoryMethod( targetType );
+            return null;
```

Comment:
Why return `null` here?

Reply:
I changed the design here a bit.. If you look at the call site: 

```java
        private MethodReference getFactoryMethod(Type returnTypeImpl, SelectionParameters selectionParameters) {
            MethodReference factoryMethod = ObjectFactoryMethodResolver.getFactoryMethod( method,
                            returnTypeImpl,
                            selectionParameters,
                            ctx
            );
            if ( factoryMethod == null && returnTypeBuilder != null ) {
                factoryMethod = ObjectFactoryMethodResolver.getBuilderFactoryMethod( method, returnTypeBuilder );
            }

            return factoryMethod;
        }
```

You'll find that there's more to check.. Error handling for not-found methods takes place at the call sites of this method. At the other hand, when there's too many qualifying methods, we know its wrong anyway...